tests/behaviour/typeql/steps.rs

```diff
@@ -20,49 +20,72 @@
  */
 
 use cucumber::{gherkin::Step, given, then, when};
-use futures::TryStreamExt;
+use futures::{TryFutureExt, TryStreamExt};
+use typeql_lang::parse_query;
 
 use crate::{behaviour::Context, generic_step_impl};
 
 generic_step_impl! {
     #[step(expr = "typeql define")]
     async fn typeql_define(context: &mut Context, step: &Step) {
-        // FIXME parse
-        context.transaction().query().define(step.docstring().unwrap()).await.unwrap();
+        let parsed = parse_query(step.docstring().unwrap());
+        assert!(parsed.is_ok());
+        let res = context.transaction().query().define(&parsed.unwrap().to_string()).await;
+        assert!(res.is_ok());
     }
 
     #[step(expr = "typeql define; throws exception")]
     async fn typeql_define_throws(context: &mut Context, step: &Step) {
-        // FIXME parse
-        assert!(context.transaction().query().define(step.docstring().unwrap()).await.is_err());
+        let parsed = parse_query(step.docstring().unwrap());
+        if parsed.is_ok() {
+            let res = context.transaction().query().define(&parsed.unwrap().to_string()).await;
+            assert!(res.is_err());
+        }
     }
 
     #[step(expr = "typeql define; throws exception containing {string}")]
     async fn typeql_define_throws_exception(context: &mut Context, step: &Step, exception: String) {
-        // FIXME parse
-        let res = context.transaction().query().define(step.docstring().unwrap()).await;
-        assert!(res.is_err());
-        assert!(res.unwrap_err().to_string().contains(&exception));
+        let result = async { parse_query(step.docstring().unwrap()).map_err(|error| error.to_string()) }
+            .and_then(|parsed| async move {
+                context.transaction().query().define(&parsed.to_string()).await.map_err(|error| error.to_string())
+            })
+            .await;
+        assert!(result.is_err());
+        assert!(result.unwrap_err().contains(&exception));
     }
 
     #[step(expr = "typeql insert")]
     async fn typeql_insert(context: &mut Context, step: &Step) {
-        // FIXME parse
-        let stream = context.transaction().query().insert(step.docstring().unwrap()).unwrap();
-        stream.try_collect::<Vec<_>>().await.unwrap();
+        let parsed = parse_query(step.docstring().unwrap());
+        assert!(parsed.is_ok());
+        let stream = context.transaction().query().insert(&parsed.unwrap().to_string());
```

Comment:
`stream` is not a very good name - it doesn't give you any context for what it represents (only the type, which the IDE should already tell you!). 
I'd say this should be 'answers' or 'inserted'. 
cc @dmitrii-ubskii  since you wrote the original version on the left

Reply:
Applies throughout

Reply:
note: `answers` is only the right name if it actually returns answers! If it's just a `Result` of nothing, then `result` is an appropriate name

Reply:
Right - I was saying that `result` is worse than `stream` which is worse than `answers` or `inserted`. However, for `define` queries for example that have a void return type and are just a checked result, `result` is the best name I think.

Reply:
Renamed to `inserted`.